lib/private/Preview/Generator.php

```diff
@@ -201,7 +203,17 @@ private function getMaxPreview(ISimpleFolder $previewFolder, File $file, $mimeTy
 				$maxWidth = (int)$this->config->getSystemValue('preview_max_x', 4096);
 				$maxHeight = (int)$this->config->getSystemValue('preview_max_y', 4096);
 
+				$preview_concurrency = (int) $this->config->getSystemValue("preview_concurrency_new", 4);
+				$sem = sem_get(Generator::SEMAPHORE_ID_NEW, $preview_concurrency);
```

Comment:
```suggestion
				$sem = extension_loaded('pcntl') && sem_get(Generator::SEMAPHORE_ID_NEW, $preview_concurrency);
```

https://docs.nextcloud.com/server/latest/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation pcntl is not a hard dependency.

Reply:
The module providing `sem_*` functions should be "`sysvsem`". It is not enabled by default according to [PHP documentation](https://www.php.net/manual/en/sem.installation.php), but is installed and enabled by default on [Debian](https://packages.debian.org/buster/amd64/php7.3-common/filelist). I guess other distros have it included in by default, too.